### PR TITLE
Emissions2

### PR DESCRIPTION
**Pull request by adfarth**
**See PR #274 for previous comments**
   ### What kind of change does this PR introduce?
(Bug fix, feature, docs update, ...)
- New feature with documentation update
## Current behavior 
- Currently, REopt calculates year 1 CO2 emissions savings from a DER investment, accounting for on-site fuel burn and grid-purchased electricity.
- REopt also currently calculates annual percent RE electricity generation as the total PV and wind generation divided by total electric load

## New Features:
We have expanded upon this RE and emissions accounting to additionally include:
- Health-related emissions (NOx, PM2.5, SO2) from grid-purchased electricity and on-site fuel burn
- Total (lifecycle) emissions (lbs) accounting (as opposed to just year 1), considering future changes in grid emissions intensity
- Total (lifecycle) emissions costs accounting, considering the social cost of CO2 and public health costs of NOx, SO2, and PM2.5
- Additional user flexibility in RE and emissions accounting to include or exclude exported RE electricity and/or the associated emissions offsets in calculating total annual RE and emissions
- Added RE heat calculations, based on RE fuel percent input for all heating techs (CHP, boiler, newboiler)
- Updated RE electricity and heat calculations to focus on load rather than generation (i.e. subtracting out curtailment, waste heat,  and storage losses)

We have also incorporated the capability for a REopt user to implement three types of clean energy goals:
- The user can set a percent renewable electricity target. This answers a question such as “How do I achieve a 25% renewable electricity target at my site?”
- The user can set a percent climate emissions reduction target. This answer a question such as “How do I reduce my site’s CO2 emissions by 50% with an investment in DERs, relative to current operations?”
- The user can include climate and/or health emissions costs in the REopt objective life cycle cost objective function. This answers questions such as “How does the cost-optimal system change if I consider the $/t costs of climate (CO2) and health (PM2.5, NOx, SO2) emissions?”

## Changed behavior:
- Year 1 CO2 emissions accounting should remain the same, with the exception of the added flexibility for a user to include or exclude exported electricity 

### Does this PR introduce a breaking change?
(What changes might users need to make in their application due to this PR?)

## TODO
- [x]  Handle "bad" Lat/Long input in validators.py - currently getting "unexpected error" in CAMx grid conversion. See https://github.com/NREL/REopt_Lite_API/pull/274#issuecomment-969325370.
- [x] Add endpoint for health costs from EASIUR 
- [x] Incorporate newboiler, steam turbine, and GHP considerations into RE and emissions calcs (kk)
- [x] Add $/tCO2 breakeven equation (kk)
- [x] Trim down test_re_emissions_constraints.py to just a few scenarios (kk)
- [x] Add endpoint for fuel emissions rates (ADF)
- [x] Add cooling elec load to overall elec load in RE elec calc (kk)
- [x] Add overall RE quantity and percent outputs (kk)

## Future TODO
- [ ] Allow user to upload custom CO2 emissions profile and ignore health impacts (rather than zeroing health out)
- [ ] Account for "outage is major event." Emissions accounting currently assumes any modeled outage occurs in every year of the analysis period (as does the energy bill calculations, etc.) 
- [ ] Allow user to decide whether or not to include outage emissions in emissions calcs?
- [ ] Option to include/exclude emissions from biogenic fuels (biofuels)
- [ ] Consider linkage between RE accounting and RECs ($/kWh value)- currently $/kWh entered as production credit by tech
- [ ] Allow renewable attribute of grid electricity and/or purchasing RECs to be considered as alternative cost-effective options
- [ ] Consider update %RE fuel input for biofuel (and other fuels?)

### Please check if the PR fulfills these requirements
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Review by Bill-Becker — APPROVED** [review]

**Review by hdunham — APPROVED** [review]

File: reo/validators.py [hunk 1]
```diff
@@ -1198,7 +1260,7 @@ def check_special_cases(self, object_name_path, template_values=None, real_value
 
                     # check if user intended to run CHP and supplied sufficient pararmeters to run CHP
                     if user_supplied_chp_inputs:
-                        required_keys = prime_mover_defaults_all['recip_engine'].keys()
+                        required_keys = prime_mover_defaults_all['recip_engine'].keys() # TODO: Check this for NOx rate
```
Comment (kkrah) [inline reo/validators.py:1263]: @adfarth is this TODO complete?

File: reo/validators.py [hunk 2]
```diff
@@ -440,7 +485,7 @@ def __init__(self, input_dict, ghpghx_inputs_validation_errors=None):
         self.input_as_none = []
         self.invalid_inputs = []
         self.resampled_inputs = []
-        self.emission_warning = []
+        self.emission_warning = [] ## I defined this the same way for NOx, might cause an error
```
Comment (kkrah) [inline reo/validators.py:488]: @adfarth same here

Comment (adfarth) [reply reo/validators.py]: Yes, this comment is outdated. Removed in [1efd149](https://github.com/NREL/REopt_Lite_API/pull/277/commits/1efd149dafd062deb009b460ccd0984a70ea9e4e)

File: reo/validators.py (hunk 1 above)
Comment (adfarth) [reply reo/validators.py]: Yes, this comment is outdated. Removed in [1efd149](https://github.com/NREL/REopt_Lite_API/pull/277/commits/1efd149dafd062deb009b460ccd0984a70ea9e4e)

**Review by kkrah — APPROVED** [review]

**Pull request merged**